flink-autoscaler/src/main/java/org/apache/flink/autoscaler/JobAutoScalerImpl.java

```diff
@@ -121,37 +120,10 @@ public void cleanup(KEY jobKey) {
     }
 
     private void clearStatesAfterAutoscalerDisabled(Context ctx) throws Exception {
-        var needFlush = false;
-        var parallelismOverrides = stateStore.getParallelismOverrides(ctx);
-        if (!parallelismOverrides.isEmpty()) {
-            needFlush = true;
-            stateStore.removeParallelismOverrides(ctx);
-        }
-
-        var collectedMetrics = stateStore.getCollectedMetrics(ctx);
-        if (!collectedMetrics.isEmpty()) {
-            needFlush = true;
-            stateStore.removeCollectedMetrics(ctx);
-        }
-
-        var scalingHistory = stateStore.getScalingHistory(ctx);
-        if (!scalingHistory.isEmpty()) {
-            var trimmedScalingHistory =
-                    trimScalingHistory(clock.instant(), ctx.getConfiguration(), scalingHistory);
-            if (trimmedScalingHistory.isEmpty()) {
-                // All scaling histories are trimmed.
-                needFlush = true;
-                stateStore.removeScalingHistory(ctx);
-            } else if (!scalingHistory.equals(trimmedScalingHistory)) {
-                // Some scaling histories are trimmed.
-                needFlush = true;
-                stateStore.storeScalingHistory(ctx, trimmedScalingHistory);
-            }
-        }
-
-        if (needFlush) {
-            stateStore.flush(ctx);
-        }
+        stateStore.removeParallelismOverrides(ctx);
+        stateStore.removeCollectedMetrics(ctx);
+        stateStore.removeScalingHistory(ctx);
+        stateStore.flush(ctx);
```

Comment:
can we add a new `clearAll` or `removeAll` method that would guarantee that all state is cleaned and then we can test for subsequent flushing not hitting k8s api

Reply:
Done.